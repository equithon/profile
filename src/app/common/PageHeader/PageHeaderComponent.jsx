import React from "react";
import styled from "styled-components";

import ColorLogo from "../../../static/img/logo/logo_tiny_color.png";
import Heading from "../Heading/HeadingComponent";

const Container = styled.div`
  height: 6vh;
  position: relative;

  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

const MainContents = styled.div`
  height: 100%;
  display: flex;

  cursor: pointer;
`;

const Logo = styled.img`
  max-height: 100%;
  margin: auto;
`;

const Title = styled(Heading)`
  margin: auto 0 auto 1vw;
`;

const ExtraContents = styled.div``;

const PageHeader = ({ title = "Equithon", onClickHandler, children }) => (
  <Container>
    <MainContents onClick={onClickHandler}>
      <Logo src={ColorLogo} />
      <Title size="med" color="primary">
        {title}
      </Title>
    </MainContents>

    <ExtraContents>{children}</ExtraContents>
  </Container>
);

export default PageHeader;
