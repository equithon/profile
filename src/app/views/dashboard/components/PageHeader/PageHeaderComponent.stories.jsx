import React from "react";
import { storiesOf } from "@storybook/react";
import PageHeader from "./PageHeaderComponent";

const logoClickHandler = () => console.log("clicked the logo");

storiesOf("PageHeader", module)
  .add("Default", () => <PageHeader logoClickHandler={logoClickHandler} />)
  .add("with Button", () => (
    <PageHeader logoClickHandler={logoClickHandler}>
      <button type="button">click me</button>
    </PageHeader>
  ));
