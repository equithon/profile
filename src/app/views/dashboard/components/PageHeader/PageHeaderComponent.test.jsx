/* eslint-disable no-undef */
import React from "react";

import PageHeader from "./PageHeaderComponent";

const logoClickHandler = () => console.log("clicked the logo");

it("should render correctly with no children", () => {
  const component = shallowWithTheme(<PageHeader />);
  expect(component).toMatchSnapshot();
});

it("should render correctly with a button", () => {
  const component = shallowWithTheme(
    <PageHeader logoClickHandler={logoClickHandler}>
      <button type="button">click me</button>
    </PageHeader>
  );

  expect(component).toMatchSnapshot();
});
