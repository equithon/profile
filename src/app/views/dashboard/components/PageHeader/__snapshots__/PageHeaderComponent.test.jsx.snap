// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no children 1`] = `undefined`;
